src/mmds/src/lib.rs

```diff
@@ -224,54 +203,44 @@ mod tests {
             .put_data(serde_json::from_str(data).unwrap())
             .unwrap();
 
-        // Test invalid request.
-        let request = b"HTTP/1.1";
-        let dummy_response = Response::new(Version::Http11, StatusCode::BadRequest);
-        assert!(parse_request(request).status() == dummy_response.status());
+        // Test resource not found.
+        let request_bytes = b"GET http://169.254.169.254/invalid HTTP/1.0\r\n\r\n";
+        let request = Request::try_from(request_bytes).unwrap();
+        let mut expected_response = Response::new(Version::Http10, StatusCode::NotFound);
+        expected_response.set_body(Body::new("Resource not found: /invalid.".to_string()));
+        let actual_response = convert_to_response(request);
+        assert!(actual_response == expected_response);
```

Comment:
nit:
```suggestion
        assert_eq!(actual_response, expected_response);
```
Applies to the entire test.